python/pyarrow/tests/test_array.py

```diff
@@ -1079,6 +1079,33 @@ def test_map_from_arrays():
             pa.int64()
         ))
 
+    # pass in null bitmap
+    result = pa.MapArray.from_arrays([0, 2, 2, 6], keys, items, pa.map_(
+        keys.type,
+        items.type),
+        mask=pa.array([False, True, False], type=pa.bool_())
+    )
+    assert result.equals(expected)
```

Comment:
Can you test this both with and without a type passed? (since that are two separate code paths where the mask is passed in cython)